examples/bri-3/src/bri/auth/strategy/jwt.strategy.ts

```diff
@@ -0,0 +1,20 @@
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { PassportStrategy } from '@nestjs/passport';
+import { Injectable } from '@nestjs/common';
+import { jwtConstants } from '../constants';
+
+//use this to check if the user holds a valid jwt
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy) {
+  constructor() {
+    super({
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+      ignoreExpiration: false,
+      secretOrKey: jwtConstants.secret,
+    });
+  }
+
+  async validate(payload: any) {
+    return { userId: payload.sub, username: payload.username };
```

Comment:
@ognjenkurtic if the id is resolvable to its controlling keys then id is enough. Otherwise, you need to indicate which public key controls the id. I prefer the former over the latter approach.

Reply:
@ognjenkurtic with all due respect, it is not, because it must be resolvable by **anyone**, and even **outside** of the BPI context since e.g. messaging between participants will be outside of the BPI context but using BpiSubject identifiers. See https://github.com/eea-oasis/baseline-standard/blob/main/core/baseline-core-v1.0-psd01.md#r41 and https://github.com/eea-oasis/baseline-standard/blob/main/core/baseline-core-v1.0-psd01.md#r42 ... we cannot assume that the BPI will be directly available for resolution of identifiers in the context of inter-party messaging.

Reply:
@ognjenkurtic thank you for the clarification. your comment makes sense now.